internal/driver/onvifdiscovery.go

```diff
@@ -209,3 +217,63 @@ func executeRawProbe(conn net.Conn, params netscan.Params) ([]onvif.Device, erro
 
 	return devices, nil
 }
+
+// makeDeviceMap creates a lookup table of existing devices by EndpointRefAddress
+func (d *Driver) makeDeviceMap() map[string]contract.Device {
+	devices := service.RunningService().Devices()
+	deviceMap := make(map[string]contract.Device, len(devices))
+
+	for _, dev := range devices {
+		if dev.Name == d.serviceName {
+			// skip control plane device
+			continue
+		}
+
+		onvifInfo := dev.Protocols[OnvifProtocol]
+		if onvifInfo == nil {
+			d.lc.Warnf("Found registered device %s without %s protocol information.", dev.Name, OnvifProtocol)
+			continue
+		}
+
+		endpointRef := onvifInfo["EndpointRefAddress"]
+		if endpointRef == "" {
+			d.lc.Warnf("Registered device %s is missing required %s protocol information: EndpointRefAddress.",
+				dev.Name, OnvifProtocol)
```

Comment:
i think there is a constant for this, right?
```suggestion
		endpointRef := onvifInfo[EndpointRefAddress]
		if endpointRef == "" {
			d.lc.Warnf("Registered device %s is missing required %s protocol information: %s.",
				dev.Name, OnvifProtocol, EndpointRefAddress)
```